src/common/config.cc

```diff
@@ -1227,6 +1227,47 @@ void md_config_t::diff(
   }
 }
 
+void md_config_t::diff_setting(const md_config_t *other,
```

Comment:
yes, as you are almost duplicating the `diff()` method, so it might be worthy a try.